From 124b77a7c86533cfe53c09351ca86f47ebed6e5c Mon Sep 17 00:00:00 2001
From: "Kumar, Kiran S" <kiran.s.kumar@intel.com>
Date: Fri, 12 Feb 2016 14:32:45 +0530
Subject: [PATCH 6/6] FOR_UPSTREAM [VPG]: drm/i915: CHT i915 KW critical issue
 fix

This patch fixes below critical klocwork issue in i915 found by
KW scan.
 - No need to check for both HIGH and LOW target_rr_type.
   Hence checking for HIGH rr type, else it will be for low rr.
   This removes the implication that there would be third value
   other than HIGH and LOW.

Change-Id: I6f7d8085fc6e10bbea2314b1b87acaf9f7aa1d4a
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-20063
Signed-off-by: Kumar, Kiran S <kiran.s.kumar@intel.com>
Reviewed-on: https://android.intel.com:443/469966
---
 drivers/gpu/drm/i915/intel_dsi_drrs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_dsi_drrs.c b/drivers/gpu/drm/i915/intel_dsi_drrs.c
index 0420093..9f00d4f 100644
--- a/drivers/gpu/drm/i915/intel_dsi_drrs.c
+++ b/drivers/gpu/drm/i915/intel_dsi_drrs.c
@@ -98,7 +98,7 @@ retry:
 
 		if (work->target_rr_type == DRRS_HIGH_RR)
 			prev_mode = drrs->connector->panel.fixed_mode;
-		else if (work->target_rr_type == DRRS_LOW_RR)
+		else
 			prev_mode = drrs->connector->panel.downclock_mode;
 
 		work->target_mode = drm_mode_duplicate(intel_encoder->base.dev,
-- 
1.9.1

